### PR TITLE
unhardcode LauncherLayoutInfo (#30)

### PR DESCRIPTION
**Pull request by warpech**
Initial, perhaps a little bit too naive implementation of #30

### CODE REVIEW

File: starcounter-include.html [hunk 1]
```diff
@@ -315,6 +315,16 @@
             return range.createContextualFragment(htmlStr);
         }
 
+        function findCompositionProvider(obj) {
```
Comment (tomalec) [inline starcounter-include.html]: Isn't it an overhead, especially for large view-models?

**Comment by warpech** [comment]
I've noticed that the tests are not passing. Will fix.

File: starcounter-include.html (hunk 1 above)
Comment (warpech) [reply starcounter-include.html]: Single call `findCompositionProvider(HTMLElement)` costs about 0.02ms, so I don't think there is a big overhead.

Anyway, we can add some caching to the function:

```js
        var cachedKey;
        function findCompositionProvider(obj) {
            if (cachedKey) {
                return obj[cachedKey];
            }
            for (var i in obj) {
                if (obj.hasOwnProperty(i) && obj[i] != null && obj[i].hasOwnProperty("PartialId")) {
                    cachedKey = i;
                    return obj[i];
                }
            }
        };
```

Comment (tomalec) [reply starcounter-include.html]: How did you measure it?

Should we also observe the changes in runtime? as theoretically the position of this property may change in fly.

I think it's worth adding benchmark as we will perform this operation 2x number of partials on page load, then on every change of property of every partial view-model.

**Comment by tomalec** [comment]
Finishing as decided https://github.com/Starcounter/starcounter-include/issues/30#issuecomment-292965760

**Pull request merged**